File: app/src/main/java/baidu/com/testlibproject/ui/NotificationActivity.java
package baidu.com.testlibproject.ui;

import android.app.Activity;
import android.app.Notification;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.Intent;
import android.support.v4.app.NotificationCompat.Builder;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import baidu.com.testlibproject.MainActivity;
import baidu.com.testlibproject.R;

public class NotificationActivity extends Activity implements View.OnClickListener {

    private static final int NOTIFICATION_ID = 0x123;

    private Button mSendBtn;
    private Button mCancelBtn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.test_notification_layout);
        initView();
    }

    private void initView() {
        mSendBtn = (Button) findViewById(R.id.send_btn);
        mCancelBtn = (Button) findViewById(R.id.cancel_btn);
        mSendBtn.setOnClickListener(this);
        mCancelBtn.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        if (v == mSendBtn) {
            sendNotification();
        } else if (v == mCancelBtn) {
            cancelNotification();
        }
    }

    private void sendNotification() {
        Intent intent = new Intent(this, MainActivity.class);
        PendingIntent pi = PendingIntent.getActivity(this, 0, intent, 0);
        Builder builder = new Builder(this);
        builder.setTicker(getString(R.string.notify_ticker));
        builder.setContentTitle(getString(R.string.notify_title));
        builder.setContentText(getString(R.string.notify_text));
        builder.setSmallIcon(R.mipmap.ic_launcher);
        builder.setContentIntent(pi);
        builder.setAutoCancel(true);
        builder.setWhen(System.currentTimeMillis());
        Notification notify = builder.build();
        NotificationManager nm = (NotificationManager) getSystemService(NOTIFICATION_SERVICE);
        nm.notify(NOTIFICATION_ID, notify);
    }

    private void cancelNotification() {
        NotificationManager nm = (NotificationManager) getSystemService(NOTIFICATION_SERVICE);
        nm.cancel(NOTIFICATION_ID);
    }
}

File: app/src/main/java/baidu/com/testlibproject/service/NotificationActivity.kt
package baidu.com.testlibproject.service

import android.app.Activity
import android.app.NotificationManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.core.app.NotificationCompat.Builder
import baidu.com.testlibproject.MainActivity
import baidu.com.testlibproject.R

class NotificationActivity : Activity(), View.OnClickListener {

    private var mSendBtn: Button? = null
    private var mCancelBtn: Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.test_notification_layout)
        initView()
    }

    private fun initView() {
        mSendBtn = findViewById(R.id.send_btn)
        mCancelBtn = findViewById(R.id.cancel_btn)
        mSendBtn!!.setOnClickListener(this)
        mCancelBtn!!.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        if (v === mSendBtn) {
            sendNotification()
        } else if (v === mCancelBtn) {
            cancelNotification()
        }
    }

    private fun sendNotification() {
        val intent = Intent(this, MainActivity::class.java)
        val pi = PendingIntent.getActivity(this, 0, intent, 0)
        val builder = Builder(this)
        builder.setTicker(getString(R.string.notify_ticker))
        builder.setContentTitle(getString(R.string.notify_title))
        builder.setContentText(getString(R.string.notify_text))
        builder.setSmallIcon(R.mipmap.ic_launcher)
        builder.setContentIntent(pi)
        builder.setAutoCancel(true)
        builder.setWhen(System.currentTimeMillis())
        val notify = builder.build()
        val nm = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        nm.notify(NOTIFICATION_ID, notify)
    }

    private fun cancelNotification() {
        val nm = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        nm.cancel(NOTIFICATION_ID)
    }

    companion object {

        private val NOTIFICATION_ID = 0x123
    }
}
